runtime/compiler/runtime/RelocationRuntime.cpp

```diff
@@ -537,6 +537,15 @@ void TR_RelocationRuntime::relocationFailureCleanup()
       }
    }
 
+static int32_t strHash(const char *str)
+   {
+   // The string hash from Java
+   int32_t result = 0;
+   for (const unsigned char *s = reinterpret_cast<const unsigned char*>(str); *s; s++)
+      result = result * 33 + *s;
```

Comment:
In Java, the multiplier is 31, not 33; see https://github.com/eclipse-openj9/openj9/blob/master/jcl/src/java.base/share/classes/java/lang/String.java#L6109 and https://github.com/eclipse-openj9/openj9/blob/master/jcl/src/java.base/share/classes/java/lang/String.java#L6119. If the intent is to produce the same hash values, this should be correct, if not, the comment should be removed.

Reply:
I think it's used more to have a unique hash value to print out to the vlog than knowing the actual hash value, though because I don't work on the inliner, I don't know that for sure. Since this came to your attention, perhaps you could open a PR (and the associated one in OMR) to clean it up?

Reply:
I could, but I need to understand whether the intent was to match Java or not: Should the multiplier be 31 and the comment stay, or should the comment be removed?

Reply:
@0xdaryl or @vijaysun-omr do you know?

Reply:
I don't have prior knowledge of this logic, so I cannot bring that kind of perspective here. Having said that, I don't see the connection between mimicing string hash semantics and printing a unique hash value in verbose JIT output. I believe (as Irwin also thought) that a unique hash value is what is intended, rather than a specific hash value, but maybe Daryl or someone else who did touch this code has a different view.

Reply:
> the associated one in OMR

@dsouzai Could you share a link to that change?